File: app/src/main/java/br/com/ibm/santander/wallacebaldenebre/ui/main/fragments/investment/InvestmentContract.java
package br.com.ibm.santander.wallacebaldenebre.ui.main.fragments.investment;

import java.util.HashMap;

import br.com.ibm.santander.wallacebaldenebre.model.Screen;
import br.com.ibm.santander.wallacebaldenebre.ui.base.MvpContract;

public interface InvestmentContract {
    interface View extends MvpContract.MvpView {
        void casts(android.view.View view);

        void setUp(android.view.View view);
    }

    interface Presenter<V extends MvpContract.MvpView> extends MvpContract.MvpPresenter<V> {
        void showDataFromServer(InvestmentFragment infra, final InvestmentCallback<HashMap<String, Screen>> callback);
    }
}

File: InvestApp/app/src/main/java/com/study/vipoliveira/investapp/ui/investment/InvestmentContract.kt
package com.study.vipoliveira.investapp.ui.investment

import com.study.vipoliveira.investapp.ui.BaseContract

interface InvestmentContract {
    interface View: BaseContract.View{

    }

    interface Presenter: BaseContract.Presenter<View> {
        fun getInvestments()
    }
}
